src/coreclr/tools/Common/Internal/NativeFormat/NativeFormatWriter.cs

```diff
@@ -195,7 +195,7 @@ public void PatchByteAt(int offset, byte value)
         // This is the price we have to pay for using UTF8. Thing like High Surrogate Start Char - '\ud800'
         // can be expressed in UTF-16 (which is the format used to store ECMA metadata), but don't have
         // a representation in UTF-8.
-        private static Encoding _stringEncoding = new UTF8Encoding(false, false);
+        private static readonly UTF8Encoding _stringEncoding = new UTF8Encoding(false, false);
```

Comment:
Nit: while you're here, `s_stringEncoding`?

Reply:
Also, I thought we had an analyzer that suggested making such fields readonly. Do we not, or is it not enabled?

Reply:
Nope https://github.com/dotnet/runtime/blob/8798c0459a36463bf3355f1059ad97fdd890c99e/eng/CodeAnalysis.src.globalconfig#L1461

Reply:
If memory serves, that analyzer has a few issues for us:
- it's for both instance and static fields, but we run into issues with private due to interop and writes that happen in native, making the readonly misleading
- it doesn't go far enough as it only targets private and we'd like internal as well

@sharwell, is there any way to configure IDE044 to only look at statics?